SleepEngine/Engine/EntityManager.cpp

```diff
@@ -0,0 +1,88 @@
+#include "stdafx.h"
+#include "EntityManager.h"
+
+BEGIN_NAMESPACE_SLEEP
+
+void EntityManager::addObject(std::unique_ptr<Object>&& object, bool shouldUpdate)
+{
+    m_addLaterObjects.emplace_back(std::move(object), shouldUpdate);
+}
+
+void EntityManager::removeObjectLater(Object const* object)
+{
+    if (!contains(m_removeLaterObjects, object))
```

Comment:
Assert

Reply:
In my opinion, it should be ok when few things want to destroy one other thing, and they all should be able to call this method (it should be legal)

Reply:
better make this explicit some how with the ability to check if queued, but that's more complex so ok